owners/test/reviewer_selection.test.js

```diff
@@ -0,0 +1,253 @@
+  const sandbox = sinon.createSandbox();
+  const ownersTree = new OwnersTree();
+
+  const rootDirRule = new OwnersRule('OWNERS.yaml', ['root']);
+  const childDirRule = new OwnersRule('foo/OWNERS.yaml', ['child']);
+  const otherChildDirRule = new OwnersRule('biz/OWNERS.yaml', ['child', 'kid']);
+  const thirdChildDirRule = new OwnersRule('buzz/OWNERS.yaml', ['thirdChild']);
+  const descendantDirRule = new OwnersRule('foo/bar/baz/OWNERS.yaml', [
+    'descendant',
+  ]);
+
+  const rootDirTree = ownersTree.addRule(rootDirRule);
+  const childDirTree = ownersTree.addRule(childDirRule);
+  const otherChildDirTree = ownersTree.addRule(otherChildDirRule);
+  const thirdChildDirTree = ownersTree.addRule(thirdChildDirRule);
+  const descendantDirTree = ownersTree.addRule(descendantDirRule);
+
+  afterEach(() => {
+    sandbox.restore();
+  });
+
+  describe('nearestOwnersTrees', () => {
+    it('returns the nearest trees', () => {
+      const fileTreeMap = ReviewerSelection.buildFileTreeMap(
+        ['./main.js', 'biz/style.css', 'foo/bar/other_file.js'],
+        ownersTree
+      );
+      const nearestTrees = ReviewerSelection._nearestOwnersTrees(fileTreeMap);
+
+      expect(nearestTrees).toEqual(
+        expect.arrayContaining([rootDirTree, otherChildDirTree, childDirTree])
+      );
+    });
+
+    it('does not return duplicates', () => {
+      const fileTreeMap = ReviewerSelection.buildFileTreeMap(
+        ['foo/file.js', 'foo/bar/other_file.js'],
+        ownersTree
+      );
+      const nearestTrees = ReviewerSelection._nearestOwnersTrees(fileTreeMap);
+
+      expect(nearestTrees).toEqual([childDirTree]);
+    });
+  });
+
+  describe('reviewersForTrees', () => {
+    it('unions the owners for all subtrees', () => {
+      const reviewers = ReviewerSelection._reviewersForTrees([
+        otherChildDirTree,
+        descendantDirTree,
+      ]);
+
+      expect(reviewers).toEqual(
+        expect.arrayContaining(['child', 'kid', 'descendant'])
+      );
+    });
+
+    it('does not include inherited owners', () => {
+      const reviewers = ReviewerSelection._reviewersForTrees([childDirTree]);
+
+      expect(reviewers).not.toContain('root');
+    });
+  });
+
+  describe('findPotentialReviewers', () => {
+    it('returns reviewers for the deepest trees', () => {
+      const fileTreeMap = ReviewerSelection.buildFileTreeMap(
+        ['./main.js', 'biz/style.css', 'foo/bar/other_file.js'],
+        ownersTree
+      );
+      sandbox.stub(ReviewerSelection, '_reviewersForTrees').callThrough();
+      const reviewers = ReviewerSelection._findPotentialReviewers(fileTreeMap);
+
+      sandbox.assert.calledWith(
+        ReviewerSelection._reviewersForTrees,
+        sinon.match.array.contains([childDirTree, otherChildDirTree])
+      );
+      expect(reviewers).toEqual(expect.arrayContaining(['child', 'kid']));
+    });
+  });
+
+  describe('filesOwnedByReviewer', () => {
+    it('lists files for which the reviewer is an owner', () => {
```

Comment:
Use an `it.each` parameterized test here - split it on the rootFiles/childFiles/kidFiles/descendantFiles part

Reply:
Done.